jerry-core/jerry.c

```diff
@@ -2115,6 +2148,11 @@ jerry_value_to_boolean (const jerry_value_t value) /**< input value */
 {
   jerry_assert_api_available ();
 
+  if (ECMA_IS_VALUE_ERROR (value))
+  {
+    return false;
+  }
+
```

Comment:
I think a value with error flag has a logical value. Or you do this for consistency?